src/proto/mgmt/storage_scm.proto

```diff
@@ -0,0 +1,77 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+syntax = "proto3";
+package mgmt;
+
+// SCM Storage Protobuf Definitions related to interactions between
+// DAOS control server and locally attached storage devices.
+
+// ScmModule represent Storage Class Memory modules installed.
+message ScmModule {
+	message Location {
+		// The channel id where module is installed.
+		uint32 channel = 1;
+		// The channel position where module is installed.
+		uint32 channelpos = 2;
+		// The memory controller id attached to module.
+		uint32 memctrlr = 3;
+		// The socket id attached to module.
+		uint32 socket = 4;
+	}
+
+	// The uid of the module.
+	//string uid = 1;
+	// The physical id of the module.
+	uint32 physicalid = 1;
+	// The device handle of the module.
+	//string handle = 3;
+	// The serial number of the module.
+	//string serial = 8;
```

Comment:
I think serial is good to include, as well as possibly the handle. Handle is the main ID used in ipmctl so if there is an issue, it would make sense to include that identifier.

Reply:
yes planning to just have to modify bindings to parse which I haven't done yet. also planning on uncommenting fwrev and uid (unless you think uid is unnecessary)

Reply:
Probably not necessary in 99% of cases, but if it doesn't really cost us anything extra to get it, it's one of the two valid identifiers, along with the handle.